src/algorithm/minkowski/gjk/mod.rs

```diff
@@ -107,11 +108,78 @@ where
                     return Some(simplex);
                 }
             }
-            i += 1;
-            if i >= MAX_ITERATIONS {
+        }
+
+        None
+    }
+
+    /// Compute the distance between the given primitives.
+    ///
+    /// ## Parameters:
+    ///
+    /// - `left`: left primitive
+    /// - `left_transform`: model-to-world-transform for the left primitive
+    /// - `right`: right primitive,
+    /// - `right_transform`: model-to-world-transform for the right primitive
+    ///
+    /// ## Returns:
+    ///
+    /// Will optionally return the distance between the objects. Will return None, if the objects
+    /// are colliding.
+    pub fn distance<P, PL, PR, TL, TR>(
```

Comment:
does it make sense for the transform types to be different ever? Or is this just generic because we can?

Reply:
For the most part users will use the same I'm sure, I just wanted to be as open minded as I could.